src/python/pose_format/bin/directory.py

```diff
@@ -1,45 +1,171 @@
+) -> Path:
+    """
+    Given a video path, and whether to keep the suffix, returns the expected corresponding path with .pose extension.
+
+    Parameters
+    ----------
+    video_path : Path
+        Path to a video file
+    keep_video_suffixes : bool, optional
+        Whether to keep suffix (e.g. foo.mp4 -> foo.mp4.pose) 
+        or replace (foo.mp4->foo.pose). Defaults to replace.
+
+    Returns
+    -------
+    Path
+        pathlib Path
+    """
+    if keep_video_suffixes:
+        return video_path.with_name(f"{video_path.name}.pose")
+    else:
+        return video_path.with_suffix(".pose")
 
 
 def main():
     parser = argparse.ArgumentParser()
-    parser.add_argument('--format',
-                        choices=['mediapipe'],
-                        default='mediapipe',
-                        type=str,
-                        help='type of pose estimation to use')
-    parser.add_argument("--directory", type=str, required=True)
-    parser.add_argument('--additional-config', type=str, help='additional configuration for the pose estimator')
+    parser.add_argument(
+        "-f",
+        "--format",
+        choices=["mediapipe"],
+        default="mediapipe",
+        type=str,
+        help="type of pose estimation to use",
+    )
+    parser.add_argument(
+        "-d",
+        "--directory",
+        type=Path,
+        required=True,
+        help="Directory to search for videos in",
+    )
+    parser.add_argument(
+        "-r",
+        "--recursive",
+        action="store_true",
+        help="Whether to search for videos recursively",
+    )
+    parser.add_argument(
+        "--keep-video-suffixes",
+        action="store_true",
+        help="Whether to drop the video extension (output for foo.mp4 becomes foo.pose, and foo.webm ALSO becomes foo.pose) or append to it (foo.mp4 becomes foo.mp4.pose, foo.webm output is foo.webm.pose). If there are multiple videos with the same basename but different extensions, this will create a .pose file for each. Otherwise only the first video will be posed.",
+    )
+    parser.add_argument(
+        "--video-suffixes",
+        type=str,
+        choices=_SUPPORTED_VIDEO_FORMATS,
+        default=_SUPPORTED_VIDEO_FORMATS,
+        help="Video extensions to search for. Defaults to searching for all supported.",
+    )
+    parser.add_argument(
+        "--additional-config",
+        type=str,
+        help="additional configuration for the pose estimator",
+    )
     args = parser.parse_args()
 
-    missing_pose_files = find_missing_pose_files(args.directory)
+    videos_with_missing_pose_files = find_videos_with_missing_pose_files(
+        args.directory,
+        video_suffixes=args.video_suffixes,
+        recursive=args.recursive,
+        keep_video_suffixes=args.keep_video_suffixes,
+    )
+
+    print(f"Found {len(videos_with_missing_pose_files)} videos missing pose files.")
+
+    pose_files_that_will_be_created = set(
```

Comment:
if you do `{... for ... in ...}` that is a set, no need to create a list and cast a set

Reply:
Huh! TIL